hypershift-operator/metrics_test.go

```diff
@@ -137,3 +147,69 @@ func TestMetrics(t *testing.T) {
 		})
 	}
 }
+
+func TestGetOperatorImage(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name            string
+		hypershiftPod   *corev1.Pod
+		expectedImage   string
+		expectedImageID string
+	}{
+		{
+			name: "When the pod is running it should get details from it",
+			hypershiftPod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "hypershift",
+				},
+				Status: corev1.PodStatus{
+					Conditions: []corev1.PodCondition{
+						{
+							Type:               corev1.PodReady,
+							Status:             corev1.ConditionTrue,
+							LastTransitionTime: metav1.Time{Time: time.Time{}.Add(10 * time.Minute)},
+						},
+					},
+					ContainerStatuses: []corev1.ContainerStatus{
+						{
+							Name:    assets.HypershiftOperatorName,
+							Image:   "quay.io/hypershift/hypershift:latest",
+							ImageID: "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+							State: corev1.ContainerState{
+								Running: &corev1.ContainerStateRunning{
+									StartedAt: metav1.Time{Time: time.Time{}},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedImage:   "quay.io/hypershift/hypershift:latest",
+			expectedImageID: "quay.io/hypershift/hypershift@sha256:746dc6b979cd6d42d4e6f3214e20eb5a2e0a2b4d1c1345f6914e5492e69d9afe",
+		},
+		{
+			name:            "When the pod is not running it should return strings with 'not found' and an error",
+			hypershiftPod:   &corev1.Pod{},
+			expectedImage:   "not found",
+			expectedImageID: "not found",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.hypershiftPod.Name = fakePodName
+			t.Setenv("MY_NAMESPACE", "hypershift")
+			t.Setenv("MY_NAME", fakePodName)
+			g := NewWithT(t)
+			client := fake.NewClientBuilder().WithScheme(api.Scheme).WithObjects(tc.hypershiftPod).Build()
+			image, imageId, err := getOperatorImage(client)
```

Comment:
The way this is right now, this unit test is not doing what you'd expect.
The fake client It's always finding the pod because you are always injecting it WithObjects(tc.hypershiftPod). 

This code never actually runs
```
				if !apierrors.IsNotFound(err) {
					g.Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("error found: %v", err.Error()))
				}
```

You could e.g. add `expectedError` to your testCases and that should be true for the second test case, then in the validation
```
if tc.expectedError {
     client = fake.NewClientBuilder().WithObjects().Build()
}

if tc.error {
      g.Expect(err).To(HaveOccurred())
       expect id to be 'not found'
       expect image to be 'not found'
}
```

Reply:
you can see examples in our *_controller_tests